gremlin-go/docker-compose.yml

```diff
@@ -17,23 +17,33 @@
 
 version: "3.4"
 services:
-  gremlin-server:
-    container_name: gremlin-server
-    image: tinkerpop/gremlin-server:3.5
+
+  gremlin-test-server:
+    container_name: gremlin-test-server
+    image: tinkerpop:gremlin-test-server
+    build:
+      context: ../
+      dockerfile: gremlin-go/Dockerfile
     ports:
-      - "8182:8182"
+      - "45940:45940"
+      - "45941:45941"
+
   gremlin-go-integration-tests:
     container_name: gremlin-go-integration-tests
     image: golang:1.17
     volumes:
       - .:/go_app
+      - ../gremlin-test/features:/gremlin-test
     environment:
-      - GREMLIN_SERVER_URL=ws://gremlin-server:8182/gremlin
+      - CUCUMBER_FEATURE_FOLDER=/gremlin-test
+      - GREMLIN_SERVER_URL=ws://gremlin-test-server:45940/gremlin
+      - GREMLIN_SERVER_BASIC_AUTH_URL=wss://gremlin-test-server:45941/gremlin
       - RUN_INTEGRATION_TESTS=true
-      - RUN_INTEGRATION_WITH_ALIAS_TESTS=false
+      - RUN_INTEGRATION_WITH_ALIAS_TESTS=true
+      - RUN_BASIC_AUTH_INTEGRATION_TESTS=true
```

Comment:
missing `TEST_TRANSACTIONS=true`
or better to rename to `RUN_TRANSACTIONS_TESTS`?

Reply:
Probably better to name to `RUN_TRANSACTIONS_TESTS`. I'll edit it locally. 

Though having all of these variables littered all over the test files feels a bit weird...and using constants and functions across different test files in the same package. Should we try putting them into a helper file or something? So it's more intuitive if we wanted to edit them?

Reply:
variable `TEST_TRANSACTIONS` also defined in `pom.xml` and used by other drivers (py, js, .net)

Reply:
Ah okay, I will change it to `TEST_TRANSACTIONS` for consistency.